app/screens/channel_info/channel_info_header.js

```diff
@@ -165,7 +165,7 @@ export default class ChannelInfoHeader extends React.PureComponent {
         });
 
         const showCustomStatus = isCustomStatusEnabled && type === General.DM_CHANNEL && customStatus?.emoji && !isCustomStatusExpired;
-        const showCustomStatusExpiry = customStatus.duration !== undefined && customStatus.duration !== CustomStatusDuration.DONT_CLEAR && isCustomStatusExpirySupported;
+        const showCustomStatusExpiry = customStatus?.duration !== undefined && customStatus?.duration !== CustomStatusDuration.DONT_CLEAR && isCustomStatusExpirySupported;
```

Comment:
no need to check against `undefined`

and it is possibly not even needed to check against `CustomStatusDuration.DONT_CLEAR`, since its value is `''` and therefore falsey in itself. Could you please try that and see if it works as well? It looks like this suggestion should be sufficient.

```suggestion
        const showCustomStatusExpiry = customStatus?.duration && isCustomStatusExpirySupported;
```